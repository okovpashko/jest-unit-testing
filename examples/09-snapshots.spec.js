import React from 'react';
import renderer from 'react-test-renderer';

import MyLink from '../lib/my-link-component';
import UserService from '../lib/user-service';


describe('Snapshots', () => {
  test('objects', () => {
    const userService = new UserService();

    const user = {
      id: 11,
      name: 'Bill',
      age: 19,
    };

    const updatedUser = userService.updateUser(user, {
      email: 'bill2example.com',
      phone: '+12345678',
      gender: 'male',
    });

    expect(updatedUser).toMatchSnapshot();
  });

  test('React component', () => {
    const component = renderer.create(
      <MyLink href="http://www.facebook.com">Facebook</MyLink>,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot()
  });
});