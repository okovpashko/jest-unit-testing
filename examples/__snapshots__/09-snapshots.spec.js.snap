// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots React component 1`] = `
<a
  className="my-link"
  href="http://www.facebook.com"
  target="_self"
>
  Facebook
</a>
`;

exports[`Snapshots objects 1`] = `
Object {
  "email": "bill2example.com",
  "gender": "male",
  "phone": "+12345678",
}
`;
